modules/db/src/main/scala/gem/dao/ObservationDao.scala

```diff
@@ -5,22 +5,32 @@ import gem.enum.Instrument
 
 import doobie.imports._
 
+import scalaz._, Scalaz._
+
 object ObservationDao {
 
   def insert(o: Observation[_]): ConnectionIO[Int] =
     sql"""
-      INSERT INTO observation (observation_id, 
-                              program_id, 
-                              observation_index, 
+      INSERT INTO observation (observation_id,
+                              program_id,
+                              observation_index,
                               title,
                               instrument)
-            VALUES (${o.id}, 
-                    ${o.id.pid}, 
-                    ${o.id.index}, 
+            VALUES (${o.id},
+                    ${o.id.pid},
+                    ${o.id.index},
                     ${o.title},
                     ${o.instrument})
     """.update.run
 
+  /** Select all the observation ids associated with the given program. */
+  def selectIds(pid: Program.Id): ConnectionIO[List[Observation.Id]] =
+    sql"""
+      SELECT observation_id
+        FROM observation
+       WHERE program_id = $pid
+    """.query[Observation.Id].list
+
   def selectFlat(id: Observation.Id): ConnectionIO[Observation[Nothing]] =
```

Comment:
What does it mean to be `Observation[Nothing]`. is that only the observation data is contained?